### DISCUSSION TITLE
Scratch windows

### DISCUSSION THREAD

**Discussion by ConsularParadi** [YaLTeR/niri#748] [Ideas]
Hi, I have been loving Niri so far. Although it is missing some xwayland features but it is way faster and lighter on resources than paperwm. As I saw on Github that it has been inspired from paperWM, I thought that scratch space would also be available but could not find anything in the documentation. Is it in the roadmap?

**Comment by YaLTeR** [comment]
Hey, scratch space on PaperWM is basically a floating layer iirc, so #122

**Reply by ConsularParadi** [reply]
It seems to be a work in progress. I'll keep an eye on it.
Thank you again for providing such an awesome WM.

**Comment by GregorySchwartz** [comment]
A scratch per window would be amazing. The current setup I have right now is, for example, a keyboard shortcut to bring a floating window, say Discord, to the current workspace centered and focused. If that scratch window does not exist then launch the program with a command, if it does exist and is on the current workspace then send it back to scratch.

**Reply by BaconIsAVeg** [reply]
> A scratch per window would be amazing. The current setup I have right now is, for example, a keyboard shortcut to bring a floating window, say Discord, to the current workspace centered and focused. If that scratch window does not exist then launch the program with a command, if it does exist and is on the current workspace then send it back to scratch.

You wouldn't mind sharing that setup would you?  I have a similiar setup for dk/leftwm using kitty and tmuxp (uses xdotool/xdo), but I'm unfamiliar how to replicate it with niri/wayland.

**Reply by GregorySchwartz** [reply]
https://github.com/GregorySchwartz/dotfiles/blob/master/.config/niri/move_window.sh

It's not perfect, but it's the best I could do for now.

**Reply by BaconIsAVeg** [reply]
Interesting.  Swapping `.is_active` for `.is_focused` is needed for multi-monitor setups, though I see the general idea is to just move the window from the current workspace to workspace 10.  Is there any flickering when it follows the application to workspace 10 then switches back to the previous one?
